smack-experimental/src/main/java/org/jivesoftware/smackx/sid/element/OriginIdElement.java

```diff
@@ -101,4 +101,25 @@ public CharSequence toXML(org.jivesoftware.smack.packet.XmlEnvironment enclosing
                 .attribute(ATTR_ID, getId())
                 .closeEmptyElement();
     }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Probably we could make use of the newly introduced `EqualsUtil` here :)

Reply:
Oh I wasn't aware of that utility class yet. Good catch!